### PR TITLE
Update Fenix release checklist

### PR DESCRIPTION
**Pull request by amedyne**
Updating this to reflect the latest process for releases for Firefox for Android.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by rvandermeulen — APPROVED** [review]
LGTM!

**Review by rocketsroger — APPROVED** [review]

**Review by csadilek — APPROVED** [review]
📝 ⭐

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -1,69 +1,37 @@
...
-
-### Bugfix uplifts / Beta Product Integrity (Beta Release until PI green signoff)
+- [ ] Review [FeatureFlags](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/FeatureFlags.kt) to determine if there are features that need to be enabled (or disabled) for Beta and Production release of Fenix. This will be a discussion with PO, PMs, EMs.
+- [ ] Make a new Beta: Follow instructions [here](https://github.com/mozilla-mobile/fenix/wiki/Creating-a-release-branch) and notify the Release Management team (slack: #releaseduty-mobile). QA team is notified that a Beta release has been captured and they will run tests for Beta release sign-off
+- [ ] Once there is GREEN QA signoff, the Release Management team (slack: #releaseduty-mobile) pushes the Beta version in the [Google Play Console](https://play.google.com/console/)
+- [ ] Check Sentry each day for issues on [Firefox Beta](https://sentry.prod.mozaws.net/operations/firefox-beta/) and if nothing concerning, Release Management team bumps releases(5%, 25%, 100%)
```
[the remark this answers is not in the document]
Comment (amedyne) [reply .github/ISSUE_TEMPLATE/release_checklist.md:21]: Thanks for clarifying @rvandermeulen

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 2]
```diff
@@ -1,69 +1,37 @@
...
 - Bugzilla account
 - Sentry access
 
-## Release Checklist
-There are two releases this covers: the current sprint that is going out to Beta, and the previous Beta that is going to Production.
-
-## Start of Sprint X.1 [Thursday, 1st week of sprint]
-- [ ] [Create an issue](https://github.com/mozilla-mobile/fenix/issues/new?template=release_checklist.md&title=Releng+for+) "Releng for v[release]" to track the current sprint.
+## Firefox for Android Release
+There are two releases this covers: the current changes in the Fenix Nightly channel that is going out to Beta, and the previous Beta that is going to Production.
```
[the remark this answers is not in the document]
Comment (amedyne) [reply .github/ISSUE_TEMPLATE/release_checklist.md:14]: 👍🏼

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 3]
```diff
@@ -1,69 +1,37 @@
...
+### Bugfix uplifts / Beta Product Integrity 
 - [ ] If bugs are considered release blocker then find someone to fix them on main and the milestone branch (cherry-pick / uplift)
-    - [ ] Add the uplift request to the appropriate row in the [Uplifts document](https://docs.google.com/spreadsheets/d/1qIvHpcQ3BqJtlzV5T4M1MhbWVxkNiG-ToeYnWEBW4-I/edit#gid=0).
-- [ ] If needed, ship a new beta version (e.g. v1.0-beta.2) and follow the submission checklist again.
-- [ ] Once there is GREEN QA signoff, file a [release management bugzilla for rollout](https://bugzilla.mozilla.org/show_bug.cgi?id=1664366)
-    - [ ] Check Sentry each day for issues on [Firefox Beta](https://sentry.prod.mozaws.net/operations/firefox-beta/) and if nothing concerning, bump release in the bugzilla (5%, 20%, 100%)
+    - [ ] Add the uplift request to the appropriate row in the [Uplifts document](https://docs.google.com/spreadsheets/d/1qIvHpcQ3BqJtlzV5T4M1MhbWVxkNiG-ToeYnWEBW4-I/edit#gid=0). Ask for approval of uplift from release owner and then notify Release Management team (slack: #releaseduty-mobile) of the uplift changes
```
[the remark this answers is not in the document]
Comment (amedyne) [reply .github/ISSUE_TEMPLATE/release_checklist.md]: Updated.

**Review by csadilek — APPROVED** [review]

**Pull request merged**